lib/Doctrine/ODM/MongoDB/SchemaManager.php

```diff
@@ -340,6 +347,124 @@ public function createSearchIndexes(): void
         }
     }
 
+    /**
+     * Wait until all search indexes are queryable for the given document classes;
+     * and all documents have been indexed.
+     *
+     * @param list<class-string>|null $classNames List of class names to check, or null to check all mapped classes
+     * @param positive-int            $maxTimeMs  Maximum time to wait in milliseconds (default: 10,000 ms)
+     * @param positive-int            $waitTimeMs Time to wait between checks in milliseconds (default: 100 ms)
+     */
+    public function waitForSearchIndexes(?array $classNames = null, int $maxTimeMs = 10_000, int $waitTimeMs = 100): void
+    {
+        if ($maxTimeMs < 1) {
+            throw new InvalidArgumentException('$maxTimeMs must be a positive number of milliseconds.');
+        }
+
+        if ($waitTimeMs < 1) {
+            throw new InvalidArgumentException('$waitTimeMs must be a positive number of milliseconds.');
+        }
+
+        $classes = $classNames === null ? $this->metadataFactory->getAllMetadata() : array_map($this->metadataFactory->getMetadataFor(...), $classNames);
+
+        // List all the collections from the Metadata with the list of their search index names
+        /** @var array<class-string, array<string, bool>> $indexesToCheck */
+        $indexesToCheck = [];
+        foreach ($classes as $class) {
+            $searchIndexes = $class->getSearchIndexes();
+            if (! $searchIndexes) {
+                continue;
+            }
+
+            $indexesToCheck[$class->getName()] = array_fill_keys(array_column($searchIndexes, 'name'), null);
+        }
+
+        $start = hrtime(true);
+        while ($indexesToCheck) {
+            if (hrtime(true) > $start + $maxTimeMs * 1_000_000) {
+                throw new MongoDBException(sprintf('Timed out waiting for search indexes to become queryable after %d ms. Search indexes are not ready for this classes: %s', $maxTimeMs, implode(', ', array_keys($indexesToCheck))));
```

Comment:
```suggestion
                throw new MongoDBException(sprintf('Timed out waiting for search indexes to become queryable after %d ms. Search indexes are not ready for the following class(es): %s', $maxTimeMs, implode(', ', array_keys($indexesToCheck))));
```

Optionally remove "the following" if you want this to be more concise.